plugins/transformations/seq2array/srcCxx/Seq2ArrayTransformation.cxx

```diff
@@ -0,0 +1,481 @@
+                                + output_union_type.member(member.name()).name().to_std_string()
+                                + ">");
+                        throw std::runtime_error(error);
+                    }
+                break;
+                // primitive type
+                default:
+                    is_compatible = is_compatible
+                            && (member.type() == output_union_type.member(member.name()).type());
+                    if (!is_compatible) {
+                        std::string error("Error: input member <"
+                                + member.name().to_std_string()
+                                + ">, not compatible with output name <"
+                                + output_union_type.member(member.name()).name().to_std_string()
+                                + ">");
+                        throw std::runtime_error(error);
+                    }
+                    break;
+                }
+
+                if (!is_compatible) {
+                    break;
+                }
+            }
+        }
+        break;
+    }
+
+    case TypeKind::ARRAY_TYPE: {
+        if (output_type.kind() != TypeKind::ARRAY_TYPE) {
+            std::string error("Error: the ouput field <" + output_type.name()
+                    + "> is different to the input type <" + input_type.name());
+            throw std::runtime_error(error);
+        }
+
+        auto input_array_type = static_cast<const ArrayType &>(input_type);
+        auto output_array_type = static_cast<const ArrayType &>(output_type);
+
+        if (input_array_type.dimension_count() != output_array_type.dimension_count()) {
+            std::string error("Error: different dimension count of input and output "
+                    "array, input name: <" + input_array_type.name()
+                    + ">, output name <" + output_array_type.name() + ">: "
+                    + std::to_string(input_array_type.dimension_count())
+                    + " != " + std::to_string(output_array_type.dimension_count()));
+            throw std::runtime_error(error);
+        }
+
+        if (input_array_type.total_element_count() != output_array_type.total_element_count()) {
+            std::string error("Error: different total element count of input "
+                    "and output array, input name: <" + input_array_type.name()
+                    + ">, output name <" + output_array_type.name() + ">: "
+                    + std::to_string(input_array_type.dimension_count())
+                    + " != " + std::to_string(output_array_type.dimension_count()));
+            throw std::runtime_error(error);
+        }
+
+        if (input_array_type.content_type() == output_array_type.content_type()) {
+            return true;
+        } else {
+            // inspect element by element since it may be compatible
+            // (output with arrays and input with sequences)
```

Comment:
This block is not inspecting "element by element". It's inspecting the type of the array's element to see if they are compatible.

Reply:
changed to "member by member" instead of "element by element", in the end, it inspects all the members within a DynamicData to see that even the non-array/seq members are the same